### ISSUE TITLE
[Parity with C#] BotAdapter UpdateActivity method signature 

### ISSUE THREAD

**Issue by gasper-az** [microsoft/botbuilder-js#1037]
**Is your feature request related to a problem? Please describe.**
We've found some differences between the `updateActivity` method from BotAdapter in DotNet and JS SDK.

In the DotNet version (link [here](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder/BotAdapter.cs#L95)), the UpdateActivityAsync method returns a _Task<[ResourceResponce](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Schema/ResourceResponse.cs)>_, which allows us to access to the _[ResourceResponse](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Schema/ResourceResponse.cs)_ properties, as its _[Id](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Schema/ResourceResponse.cs#L48)_. However, in the JS version (link [here](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder-core/src/botAdapter.ts#L39)), the method returns a _Promise<void>_, which is a completely different behavior from the DotNet's implementation.

**Describe the solution you'd like**
Add a new implementation of the _updateActivity_ in [BotAdapter.ts](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder-core/src/botAdapter.ts) which returns a _Promise<[ResourceResponse](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botframework-schema/src/index.ts#L582)>_ for ensuring parity with the [DotNet version](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder/BotAdapter.cs#L95).

**Additional context**
DotNet implementation:
![image](https://user-images.githubusercontent.com/43762887/60545143-bdafa600-9cf0-11e9-9e7d-076cd530d5a9.png)

JS implementation:
![image](https://user-images.githubusercontent.com/43762887/60545170-ca33fe80-9cf0-11e9-86b3-e6ce2439ed40.png)

[enhancement]

**Comment by stevengum** [comment]
This is a breaking change:

```bash
src/testAdapter.ts(139,12): error TS2416: Property 'updateActivity' in type 'TestAdapter' is not assignable to the same property in base type 'BotAdapter'.
  Type '(context: TurnContext, activity: Partial<Activity>) => Promise<void>' is not assignable to type '(context: TurnContext, activity: Partial<Activity>) => Promise<ResourceResponse>'.
    Type 'Promise<void>' is not assignable to type 'Promise<ResourceResponse>'.
      Type 'void' is not assignable to type 'ResourceResponse'.
src/testAdapter.ts(188,54): error TS2345: Argument of type 'this' is not assignable to parameter of type 'BotAdapter'.
  Type 'TestAdapter' is not assignable to type 'BotAdapter'.
    Types of property 'updateActivity' are incompatible.
      Type '(context: TurnContext, activity: Partial<Activity>) => Promise<void>' is not assignable to type '(context: TurnContext, activity: Partial<Activity>) => Promise<ResourceResponse>'.
```

But the BotAdapter should have returned a ResourceResposne to begin with ([swagger](https://github.com/microsoft/botframework-sdk/blob/master/specs/botframework-protocol/botframework-channel.json#L389-L414)).

We'll have to update every Adapter and break community TypeScript adapters as well ([example twilio whatsapp adapter](https://github.com/BotBuilderCommunity/botbuilder-community-js/blob/master/libraries/botbuilder-adapter-twilio-whatsapp/src/twilioWhatsAppAdapter.ts#L124-L126)).

**Comment by stevengum** [comment]
We might have to make a new "updateActivity" method that returns the correct value and mark the original `updateActivity()` as deprecated.

**Comment by stevengum** [comment]
I'll create a new method and mark the old method (`updateActivity`) as deprecated to prevent breaking users and libraries that use TypeScript and the old method.

**Comment by stevengum** [comment]
So I thought I updated this issue and #1184...

Creating a new method to return a ResourceResponse for an UpdateActivity-type operation doesn't add utility to the SDK. After testing, the ResourceResponse returned for UpdateActivity contains the Id of the activity that was just updated. You cannot update an activity without its Id, so the return value doesn't add any value.

As such, we've decided to not increase our surface area nor will we implement any breaking changes.

Closing issue.
